lib/zipkin-tracer/faraday/zipkin-tracer.rb

```diff
@@ -19,30 +19,28 @@ def initialize(app, service_name = nil)
     end
 
     def call(env)
+      @tracer = Trace.tracer
       # handle either a URI object (passed by Faraday v0.8.x in testing), or something string-izable
       url = env[:url].respond_to?(:host) ? env[:url] : URI.parse(env[:url].to_s)
-      local_endpoint = ::Trace.default_endpoint # The rack middleware set this up for us.
+      local_endpoint = Trace.default_endpoint # The rack middleware set this up for us.
       remote_endpoint = callee_endpoint(url, local_endpoint.ip_format) # The endpoint we are calling.
 
       response = nil
-      begin
-        trace_id = ::Trace.id
-        ::Trace.push(trace_id.next_id)
+      @trace_id = Trace.id.next_id # Use a var instead of Trace.id for speed
```

Comment:
and why do you use an instance var, as opposed to a local var (e.g. without the `@` prefix)?